python313Packages.atopile: 0.2.69 -> 0.11.2

Sigmanificient opened the pull request:
- #392603
- #432205

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Sigmanificient committed:
python313Packaes.atopile: 0.2.69 -> 0.3.20

changelog: https://github.com/atopile/atopile/releases/tag/v0.3.20

Sigmanificient commented:
`python3Packages.nanobind` fix is part of this pr:
- #393250

Sigmanificient committed:
python313Packages.atopile-easyda2ato: 0.9.4 -> 0.9.5

changelog:
https://github.com/atopile/easyeda2kicad.py/releases/tag/v0.9.5

Sigmanificient committed:
python313Packages.fastapi-github-oidc: init at 0.3.0

changelog:
https://github.com/atopile/fastapi-github-oidc/releases/tag/0.3.0

Sigmanificient committed:
python313Packages.atopile: 0.3.24 -> 0.4.1

changelog: https://github.com/atopile/atopile/releases/tag/v0.4.1

Sigmanificient committed:
python313Packages.kicadcliwrapper: 1.0.4 -> 1.10

changelog:https://github.com/atopile/kicadcliwrapper/releases/tag/v1.1.0

Sigmanificient committed:
python313Packages.atopile: 0.4.1 -> 0.6.0

changelog:
- https://github.com/atopile/atopile/releases/tag/v0.5.0
- https://github.com/atopile/atopile/releases/tag/v0.5.1
- https://github.com/atopile/atopile/releases/tag/v0.6.0

mweinelt commented:
Please read https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#writing-good-commit-messages and update the commit messages. We set `meta.changelog` for good reason and we're in breaking changes freeze right now.

Sigmanificient commented:
The PR has been opened for a bit more than a month, I couldn't know it would be still opened during the release breaking changes freeze :sweat_smile: I don't think it hurts to let it be reviewed and merge at the appropriate time

Sigmanificient committed:
python313Packages.kicadcliwrapper: init at 1.0.4

Sigmanificient committed:
python313Packages.atopile-easy2kicad: init at 0.9.4

Sigmanificient committed:
python313Packages.atopile: 0.6.0 -> 0.7.0

https://github.com/atopile/atopile/releases/tag/v0.7.0

Sigmanificient committed:
python313Packages.atopile: 0.7.0 -> 0.8.2

https://github.com/atopile/atopile/releases/tag/v0.8.0
https://github.com/atopile/atopile/releases/tag/v0.8.1
https://github.com/atopile/atopile/releases/tag/v0.8.2

pbsds commented on pkgs/development/python-modules/kicadcliwrapper/default.nix:
```diff
@@ -0,0 +1,55 @@
+    "kicadcliwrapper.lib"
+  ];
+
+  # this script is used the generated the bindings
+  # and is intended for development.
```
Also it is installed to `python3.12/site-packages/kicadcliwrapper/main.py`, is this intended?

Sigmanificient commented on pkgs/development/python-modules/atopile/default.nix:
```diff
@@ -2,59 +2,67 @@
-  cattrs,
-  click,
-  deepdiff,
-  easyeda2ato,
```
well i replaced it in this diff?

pbsds replied:
But `easyeda2ato` is still left in the nixpkgs tree and is essentially a duplicate of atopile-easyeda2kicad. The [pypi page](https://pypi.org/project/easyeda2ato/) even points to the same repo as atopile-easyeda2kicad

Sigmanificient committed:
python313Packages.atopile: 0.8.2 -> 0.9.0

pbsds commented:
I unresolved three unaddressed review comment

Sigmanificient committed:
python313Packages.easyda2ato: remove

Sigmanificient commented:
This is a typing thing that has been added in python 3.13: https://peps.python.org/pep-0696/

Sigmanificient committed:
python313Packages.atopile: disable python 3.12

Sigmanificient commented:
@pbsds I marked it has disabled for python 3.12 because upstream does not support it

Sigmanificient committed:
python313Packages.atopile: 0.9.0 -> 0.9.6

https://github.com/atopile/atopile/releases/tag/v0.9.1
https://github.com/atopile/atopile/releases/tag/v0.9.2
https://github.com/atopile/atopile/releases/tag/v0.9.3
https://github.com/atopile/atopile/releases/tag/v0.9.4
https://github.com/atopile/atopile/releases/tag/v0.9.5
https://github.com/atopile/atopile/releases/tag/v0.9.6

Sigmanificient committed:
python313Packages.atopile: fix posthog telemery initialization

Sigmanificient committed:
python313Packages.atopile: 0.9.6 -> 0.10.15

https://github.com/atopile/atopile/compare/v0.9.6...v0.10.15

Sigmanificient committed:
python313Packages.kicadcliwrapper: 1.1.0 -> 1.1.1

Sigmanificient committed:
python313Packages.protoletariat: init at 3.3.10

protoletariat

Sigmanificient committed:
python313Packages.kicad-python: init at 0.4.0

Sigmanificient committed:
python313Packages.atopile: 0.10.15 -> 0.10.23

https://github.com/atopile/atopile/compare/v0.10.15...v0.10.23

python313Packages.atopile: remove posthog patch

Sigmanificient committed:
python313Packages.apotile: enable tests

Sigmanificient committed:
python313Packages.atopile: 0.10.23 -> 0.11.2

SuperSandro2000 commented on pkgs/development/python-modules/protoletariat/default.nix:
```diff
@@ -0,0 +1,63 @@
+  };
+
+  build-system = [
+    setuptools
```
```suggestion
```
https://github.com/search?q=repo%3Acpcloud%2Fprotoletariat%20setuptools&type=code

SuperSandro2000 reviewed (commented):
otherwise LGTM

Sigmanificient commented:
@SuperSandro2000 Done!

SuperSandro2000 reviewed (approved):
running a nixpkgs-review just to make sure and then going to merge this finally :tada:

SuperSandro2000 commented on pkgs/development/python-modules/atopile/default.nix:
```diff
@@ -64,59 +80,146 @@ buildPythonPackage rec {
+
+  disabledTests = [
+    # timeout
+    "test_build_error_logging"
```
```suggestion
    "test_build_error_logging"
    "test_performance_mifs_bus_params"
```

Sigmanificient commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 393261`

Logs: https://github.com/Sigmanificient/nixpkgs-review-gha/actions/runs/16893248516


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>python312Packages.atopile-easyeda2kicad</li>
    <li>python312Packages.atopile-easyeda2kicad.dist</li>
    <li>python312Packages.fastapi-github-oidc</li>
    <li>python312Packages.fastapi-github-oidc.dist</li>
    <li>python312Packages.kicad-python</li>
    <li>python312Packages.kicad-python.dist</li>
    <li>python312Packages.kicadcliwrapper</li>
    <li>python312Packages.kicadcliwrapper.dist</li>
    <li>python312Packages.protoletariat</li>
    <li>python312Packages.protoletariat.dist</li>
    <li>python313Packages.atopile</li>
    <li>python313Packages.atopile-easyeda2kicad</li>
    <li>python313Packages.atopile-easyeda2kicad.dist</li>
    <li>python313Packages.atopile.dist</li>
    <li>python313Packages.fastapi-github-oidc</li>
    <li>python313Packages.fastapi-github-oidc.dist</li>
    <li>python313Packages.kicad-python</li>
    <li>python313Packages.kicad-python.dist</li>
    <li>python313Packages.kicadcliwrapper</li>
    <li>python313Packages.kicadcliwrapper.dist</li>
    <li>python313Packages.protoletariat</li>
    <li>python313Packages.protoletariat.dist</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>python312Packages.atopile-easyeda2kicad</li>
    <li>python312Packages.atopile-easyeda2kicad.dist</li>
    <li>python312Packages.fastapi-github-oidc</li>
    <li>python312Packages.fastapi-github-oidc.dist</li>
    <li>python312Packages.kicad-python</li>
    <li>python312Packages.kicad-python.dist</li>
    <li>python312Packages.kicadcliwrapper</li>
    <li>python312Packages.kicadcliwrapper.dist</li>
    <li>python312Packages.protoletariat</li>
    <li>python312Packages.protoletariat.dist</li>
    <li>python313Packages.atopile</li>
    <li>python313Packages.atopile-easyeda2kicad</li>
    <li>python313Packages.atopile-easyeda2kicad.dist</li>
    <li>python313Packages.atopile.dist</li>
    <li>python313Packages.fastapi-github-oidc</li>
    <li>python313Packages.fastapi-github-oidc.dist</li>
    <li>python313Packages.kicad-python</li>
    <li>python313Packages.kicad-python.dist</li>
    <li>python313Packages.kicadcliwrapper</li>
    <li>python313Packages.kicadcliwrapper.dist</li>
    <li>python313Packages.protoletariat</li>
    <li>python313Packages.protoletariat.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python312Packages.atopile-easyeda2kicad</li>
    <li>python312Packages.atopile-easyeda2kicad.dist</li>
    <li>python312Packages.fastapi-github-oidc</li>
    <li>python312Packages.fastapi-github-oidc.dist</li>
    <li>python312Packages.kicad-python</li>
    <li>python312Packages.kicad-python.dist</li>
    <li>python312Packages.protoletariat</li>
    <li>python312Packages.protoletariat.dist</li>
    <li>python313Packages.atopile-easyeda2kicad</li>
    <li>python313Packages.atopile-easyeda2kicad.dist</li>
    <li>python313Packages.fastapi-github-oidc</li>
    <li>python313Packages.fastapi-github-oidc.dist</li>
    <li>python313Packages.kicad-python</li>
    <li>python313Packages.kicad-python.dist</li>
    <li>python313Packages.protoletariat</li>
    <li>python313Packages.protoletariat.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python312Packages.atopile-easyeda2kicad</li>
    <li>python312Packages.atopile-easyeda2kicad.dist</li>
    <li>python312Packages.fastapi-github-oidc</li>
    <li>python312Packages.fastapi-github-oidc.dist</li>
    <li>python312Packages.kicad-python</li>
    <li>python312Packages.kicad-python.dist</li>
    <li>python312Packages.protoletariat</li>
    <li>python312Packages.protoletariat.dist</li>
    <li>python313Packages.atopile-easyeda2kicad</li>
    <li>python313Packages.atopile-easyeda2kicad.dist</li>
    <li>python313Packages.fastapi-github-oidc</li>
    <li>python313Packages.fastapi-github-oidc.dist</li>
    <li>python313Packages.kicad-python</li>
    <li>python313Packages.kicad-python.dist</li>
    <li>python313Packages.protoletariat</li>
    <li>python313Packages.protoletariat.dist</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/development/python-modules/kicad-python/default.nix:
```diff
@@ -0,0 +1,70 @@
+    pynng
+    mypy-protobuf
+  ]
+  ++ (lib.optional (pythonOlder "3.13") typing-extensions);
```
```suggestion
  ++ lib.optional (pythonOlder "3.13") typing-extensions;
```
